workflow/controller/operator.go

```diff
@@ -1523,6 +1523,21 @@ func getChildNodeIndex(node *wfv1.NodeStatus, nodes wfv1.Nodes, index int) *wfv1
 	return &lastChildNode
 }
 
+func getRetryNodeChildrenIds(node *wfv1.NodeStatus, nodes wfv1.Nodes) []string {
+	// A fulfilled Retry node will always reflect the status of its last child node, so its individual attempts don't interest us.
+	// To resume the traversal, we look at the children of the last child node and of any on exit nodes.
+	var childrenIds []string
+	for i := -1; i >= -len(node.Children); i-- {
+		node := getChildNodeIndex(node, nodes, i)
+		if strings.HasSuffix(node.Name, ".onExit") {
```

Comment:
Great catch!

Reply:
We technically don't, because the only time `getChildNodeIndex` returns `nil` is when `len(node.Children) == 0`, but since this is inside the for loop guarded by that it should never happen. However, `getChildNodeIndex` could change under us so it's always safer to add a nil check.